src/main/java/org/opensrp/service/StockService.java

```diff
@@ -404,7 +406,7 @@ private void validateFields(Inventory inventory) {
 			throw new IllegalArgumentException(SERVICE_POINT_DOES_NOT_EXISTS);
 		} else if (deliveryDate.getTime() > new Date().getTime()) {
 			throw new IllegalArgumentException(INVALID_DELIVERY_DATE);
-		} else if (!isWholeNumber(String.valueOf(inventory.getQuantity())) || inventory.getQuantity() < 1) {
+		} else if (inventory.getQuantity() != null && (!isWholeNumber(String.valueOf(inventory.getQuantity())) || inventory.getQuantity() < 1)) {
```

Comment:
inventory should always have quantity, there is now way you can have inventory without quantity.

This needs to return an error also when quantity is null

Reply:
If the quantity field is mandatory, then I believe this issue is invalid.
Since quantity is defined by an **int** datatype, if it is not provided in the payload, then it is considered as 0 (default value), and based on that value code throws a 404 error if it is less than 1 @githengi

Reply:
I believe this field was optional initially https://github.com/OpenSRP/opensrp-server-web/issues/557#issue-714545899
@cafootitt  can you please confirm this.

Reply:
Yes, it was optional. There is no logic or anything that depends on this quantity field. So if it's blank, or null, that doesn't impact anything in the user workflow, because the Quantity field is not shown in the app in that case. 

I need to understand if there's a reason on the server end that quantity cannot be optional. I'm not against making it required, but there is no user requirement for this. We can discuss on the standup today.

Reply:
As discussed in today's standup, keep the quantity field as optional. But give it a default value of 1 while persistence if it is provided as null in the payload.